contrib/win32/win32compat/w32-doexec.c

```diff
@@ -238,6 +243,43 @@ setup_session_env(struct ssh *ssh, Session* s)
 	return ret;
 }
 
+void do_loginmsg_windows(int fd) {
+	int r;
+	char* loginmsg_pipe = NULL;
+	if (sshbuf_len(loginmsg) != 0) {
```

Comment:
What's the behavior with and without this?
I would like to understand loginmsg content and who populates them.

Reply:
I think we can probably remove this function.

When I debug, loginmsg is NULL, so with or without this, no message would be printed. 

There's some places in the codebase that populate loginmsg, but I don't think we hit any of them, either due to #ifdef directives, or just not utilizing the functions:
1. In sshlogin.c, with the "last login at <time>"
2. In auth files, like auth-passwd.c with a warning about password expiration